app/code/Magento/Inventory/Model/SourceItem/Command/SourceItemsDelete.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\SourceItem\Command;
+
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\Framework\Exception\InputException;
+use Magento\Inventory\Model\ResourceModel\SourceItem\DeleteMultiple;
+use Magento\InventoryApi\Api\SourceItemsDeleteInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class SourceItemsDelete implements SourceItemsDeleteInterface
+{
+    /**
+     * @var DeleteMultiple
+     */
+    private $deleteMultiple;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param DeleteMultiple $deleteMultiple
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        DeleteMultiple $deleteMultiple,
+        LoggerInterface $logger
+    ) {
+        $this->deleteMultiple = $deleteMultiple;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $sourceItems)
+    {
+        if (empty($sourceItems)) {
+            throw new InputException(__('Input data is empty'));
+        }
```

Comment:
not sure that we need to throw an exception when someone tries to delete empty array of source Items. 

For example, you have a replace strategy and you want to delete existing SourceItems and substitute with new ones, but on clean Magento installation with empty DB - delete would have an empty set